IdSSLOpenSSLHeaders.pas

```diff
@@ -19016,6 +19210,13 @@ function IsOpenSSL_DTLSv1_Available : Boolean;
     Assigned(DTLSv1_client_method);
 end;
 
+function IsOpenSSL_TLSv1_3_Available : Boolean;
+{$IFDEF USE_INLINE} inline; {$ENDIF}
+begin
+  Result := Assigned(TLS_method) and
+    Assigned(TLSv1_method);

```

Comment:
Why is this checking for the existence of the TLS 1.0 method?

Reply:
Because that method does not exist in OpenSSL 1.0.2.

Reply:
> Because that method does not exist in OpenSSL 1.0.2.

Exactly my point. This new `IsOpenSSL_TLSv1_3_Available()` function should not be looking for any TLS 1.0-1.2 functions, only for TLS 1.3 functions.  `TLSv1_method()` is for TLS 1.0 only and is already handled by `IsOpenSSL_TLSv1_0_Available()`.  `IsOpenSSL_TLSv1_3_Available()` should be looking for `TLS_method()`, `TLS_server_method()`, and `TLS_client_method()` instead.  And in fact, since these new functions are version-flexible, the name `IsOpenSSL_TLSv1_3_Available` is a little misleading going forward when newer TLS versions are eventually released in the future.